api/payloads/service_binding.go

```diff
@@ -16,18 +18,40 @@ type ServiceBindingCreate struct {
 }
 
 func (p ServiceBindingCreate) ToMessage(spaceGUID string) repositories.CreateServiceBindingMessage {
+	var appGUID string
+	if p.Relationships.App != nil {
+		appGUID = p.Relationships.App.Data.GUID
+	}
+
 	return repositories.CreateServiceBindingMessage{
 		Name:                p.Name,
 		ServiceInstanceGUID: p.Relationships.ServiceInstance.Data.GUID,
-		AppGUID:             p.Relationships.App.Data.GUID,
+		AppGUID:             appGUID,
 		SpaceGUID:           spaceGUID,
+		Type:                p.Type,
 	}
 }
 
 func (p ServiceBindingCreate) Validate() error {
 	return jellidation.ValidateStruct(&p,
-		jellidation.Field(&p.Type, validation.OneOf("app")),
-		jellidation.Field(&p.Relationships, jellidation.NotNil),
+		jellidation.Field(&p.Type, validation.OneOf("app", "key")),
```

Comment:
We should also validate the `ServiceBindingCreate.Name` field. According to the cf api [spec](https://v3-apidocs.cloudfoundry.org/version/3.159.0/index.html#create-a-service-credential-binding) it is required for bindings of type`key` and optional for bindings of type `app`. So... some more conditional validation. I can see that right now the `Name` [field](https://github.com/ddraganovv/korifi/blob/f45b42c2f4d67f302dded849e40567f91952531b/api/payloads/service_binding.go#L17) is of type `*string`, but  I guess we should change it to a value and some conditional validation here.... unfortunately.